Examples/test-suite/preproc_defined.i

```diff
@@ -123,3 +123,6 @@ void another_macro_checking(void) {
 #if 0
 # wobble wobble
 #endif
+
+#define __attribute__(x)
+%import <preproc_defined.h>
```

Comment:
You shouldn't need to add a separate header - just put its contents here wrapped in `%inline` like so:

```
%inline %{
// put header contents here
%}
```

I suspect that'll fix your testsuite problem too - `%import` parses the file and gathers type information but doesn't actually wrap anything in the file.